cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -203,8 +240,14 @@ func (c *machineController) findScalableResourceByProviderID(providerID normaliz
 		return nil, err
 	}
 
-	if machine == nil {
-		return nil, nil
+	if c.machinePoolsAvailable {
+		machinePool, err := c.findMachinePoolByProviderID(providerID)
+		if err != nil {
+			return nil, err
+		}
+		if machinePool != nil {
+			return machinePool, nil
+		}
```

Comment:
I agree checking MachinePool last makes more sense; I changed it.

Reply:
...aand I'm wrong--the code after this clause will return (nil, nil) if it can't find a MachineSet owner, so there was a reason to structure it this way.